test/py/test_dma_unmap.py

```diff
@@ -29,33 +29,45 @@
 #
 
 import errno
+from unittest.mock import patch
 from libvfio_user import *
 
 ctx = None
 sock = None
 
 
-def test_dma_unmap_setup():
+def setup_function(function):
     global ctx, sock
-
     ctx = prepare_ctx_for_dma()
     assert ctx is not None
-    payload = struct.pack("II", 0, 0)
-
     sock = connect_client(ctx)
 
-    payload = vfio_user_dma_map(argsz=len(vfio_user_dma_map()),
-        flags=(VFIO_USER_F_DMA_REGION_READ |
-               VFIO_USER_F_DMA_REGION_WRITE),
-        offset=0, addr=0x1000, size=4096)
 
-    msg(ctx, sock, VFIO_USER_DMA_MAP, payload)
+def teardown_function(function):
+    global ctx, sock
+    disconnect_client(ctx, sock)
+    vfu_destroy_ctx(ctx)
 
 
-def test_dma_unmap_short_write():
+def with_dma_map(dma_maps=[(0x0, 0x1000)]):
+    """Function decorator that initializes the device as a PCI device."""
```

Comment:
why does this need to be a decorator?

Reply:
It doesn't. It just keeps the "envrironmet" of the test separate from the body of the test, thus making reading the test easier (at least to me). Shall I make it a standard function?

Reply:
yes please, unless it's going to wrap (prelude and postlude) it just obscures what's going IMO, and it's basically the same number of lines.

Reply:
ok